illumos-utils/src/opte/non_illumos.rs

```diff
@@ -43,3 +70,268 @@ pub fn delete_all_xde_devices(log: &Logger) -> Result<(), Error> {
     slog::warn!(log, "`xde` driver is a fiction on non-illumos systems");
     Ok(())
 }
+
+#[derive(Debug)]
+pub(crate) struct RouteInfo {
+    pub dest: IpCidr,
+    pub target: RouterTarget,
+    pub class: RouterClass,
+}
+
+impl PartialEq for RouteInfo {
+    fn eq(&self, other: &Self) -> bool {
+        if self.dest != other.dest {
+            return false;
+        }
+        match (self.class, other.class) {
+            (RouterClass::System, RouterClass::Custom) => return false,
+            (RouterClass::Custom, RouterClass::System) => return false,
+            (_, _) => {}
```

Comment:
This seems a little dicey (we won't catch new router class variants in the future). Can we derive `PartialEq` instead?

Reply:
Yeah, that's better, thanks. Will do.

Reply:
Oh! I do remember. `RouterTarget` and `RouterClass` are in OPTE, and they don't implement `PartialEq`. I'll still rework this to better handle new variants.

Reply:
I spelled out all the extant variants in 2024b245c